ML-gitops-RN-1-9-1: Documented the GitOps 1.9.1 RNs. (#62126)
openshift/openshift-docs: modules/gitops-release-notes-1-9-1.adoc

// Module included in the following assembly:
//
// * gitops/gitops-release-notes.adoc

:_content-type: REFERENCE

[id="gitops-release-notes-1-9-1_{context}"]
= Release notes for {gitops-title} 1.9.1

{gitops-title} 1.9.1 is now available on {product-title} 4.12 and 4.13.

[id="errata-updates-1-9-1_{context}"]
== Errata updates

> === RHSA-2023:3591- {gitops-title} 1.9.1 security update advisory 

Reviewer: is this the correct RHSA #? I tried going to it and it's giving me a security advisory for python3

Author: I saw that too, but i just got what was provided.... can we please get confirmation.

Reviewer 2: @iam-veeramalla, @reginapizza , @varshab1210:

- For 1.9.1, Please confirm that it is ok to add [RHSA-2023:3591](https://access.redhat.com/errata/RHSA-2023:3591) - 4-digit ID in the Errata Updates section of the RN.
- Are we including this also - https://errata.engineering.redhat.com/advisory/116282?
For [116282](https://errata.engineering.redhat.com/advisory/116282), the only safe option I can see now is that wait till the errata state is changed to SHIPPED LIVE, and then 
@masauso-lungu check the errata advisory history and title for the changed number. If it is changed then please update the RN Errata updates section correctly, get the QE approval, and then send the RN PR for the merge.

@iam-veeramalla, @reginapizza , @varshab1210: WDYT? Please let me know. Thanks!

Author: Noted!

> Issued: 2023-07-10

Reviewer: this will be 2023-07-17

Author: I had missed that.... Thank you.

The list of security fixes that are included in this release is documented in the following advisory:

> * link:https://access.redhat.com/errata/RHSA-2023:3591[RHSA-2023:3591]

Reviewer: same comment about RHSA #

Author: To be updated once we have the required link

If you have installed the {gitops-title} Operator, run the following command to view the container images in this release:

[source,terminal]
----
$ oc describe deployment gitops-operator-controller-manager -n openshift-operators
----

[id="new-features-1-9-1_{context}"]
== New features

The current release adds the following improvements:

* With this update, the bundled Argo CD has been updated to version 2.7.6.

[id="fixed-issues-1-9-1_{context}"]
== Fixed issues

The following issues have been resolved in the current release:

> * Before this update, some users were observing that Argo CD was becoming unresponsive with an increase in namespaces and applications. This update fixes the issue by removing a deadlock happening when two functions were competing for mutexes. Now, users should not experience crashes or unresponsiveness when they have a large number of resources. link:https://issues.redhat.com/browse/GITOPS-2782[GITOPS-2782]

Reviewer 2: ```suggestion
* Before this update, Argo CD was becoming unresponsive when there was an increase in namespaces and applications. This update fixes the issue by removing a deadlock. The deadlock occurs when two functions are competing for mutual exclusion locks. Now, users should not experience crashes or unresponsiveness when there is an increase in the number of resources. link:https://issues.redhat.com/browse/GITOPS-2782[GITOPS-2782]
```

> * Before this update, some users were observing that Argo CD was becoming unresponsive with an increase in namespaces and applications. This update fixes the issue by removing a deadlock happening when two functions were competing for mutexes. Now, users should not experience crashes or unresponsiveness when they have a large number of resources. link:https://issues.redhat.com/browse/GITOPS-2782[GITOPS-2782]

Reviewer: You could probably just simplify it by saying they were competing for resources.

> * Before this update, Argo CD was becoming unresponsive when there was an increase in namespaces and applications. This update fixes the issue by removing a deadlock. The deadlock occurs when two functions are competing for resources. Now, users should not experience crashes or unresponsiveness when there is an increase in the number of namespaces or applications. link:https://issues.redhat.com/browse/GITOPS-2782[GITOPS-2782]

Reviewer 3: ```suggestion
* Before this update, Argo CD was becoming unresponsive when there was an increase in namespaces and applications. This update fixes the issue by removing a deadlock. The deadlock occurs when two functions are competing for resources. Now, users should not experience crashes or unresponsiveness when there is an increase in namespaces or applications. link:https://issues.redhat.com/browse/GITOPS-2782[GITOPS-2782]
```

Reviewer 3: `the number of` is not used in the first sentence, so it seems redundant here from the minimalism point of view.

> * Before this update, users reported that `argocd-application-controller` would suddenly stop working when re-syncing. This update fixes the issue by adding logic to prevent a cluster cache deadlock. Now, users should no longer face this deadlock situation and applications should resync successfully. link:https://issues.redhat.com/browse/GITOPS-2880[GITOPS-2880]

Reviewer 2: Refer to https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting and carefully handle the following instances:

-  `argocd-application-controller`
- `known_host` (the succeeding RN text)
- `argocd-ssh-knownhosts-cm` (the succeeding RN text)
-  configmap
- appSync setting

> * Before this update, `argocd-application-controller` would suddenly stop working when resychronizing. This update fixes the issue by adding logic to prevent a cluster cache deadlock. Now, users should not experience the deadlock situation and applications should resynchronize successfully. link:https://issues.redhat.com/browse/GITOPS-2880[GITOPS-2880]

Reviewer 3: ```suggestion
* Before this update, the Argo CD application controller could stop working when resychronizing applications. This update fixes the issue by adding logic to prevent a cluster cache deadlock. Now, users should not experience the deadlock situation and applications should resynchronize successfully. link:https://issues.redhat.com/browse/GITOPS-2880[GITOPS-2880]
```

Reviewer 3: I think it would be better to use `could` instead of `would` to make it clear that it does not happen every time. Also, it should be specified what the controller is resynchronizing (the addition of word "applications").

> * Before this update, There was an RSA key mismatch for `known_host` in `argocd-ssh-knownhosts-cm`. This update fixes the issue by matching the RSA key with the upstream project. Now, users should be able to use the default RSA keys on default deployments. link:https://issues.redhat.com/browse/GITOPS-3042[GITOPS-3042]

Reviewer 3: ```suggestion
* Before this update, there was an RSA key mismatch for known hosts in the `argocd-ssh-known-hosts-cm` config map. This update fixes the issue by matching the RSA key with the upstream project. Now, users should be able to use the default RSA keys on default deployments. link:https://issues.redhat.com/browse/GITOPS-3042[GITOPS-3042]
```

> * Before this update, there was an RSA key mismatch for known hosts in the `argocd-ssh-known-hosts-cm`. This update fixes the issue by matching the RSA key with the upstream project. Now, users should be able to use the default RSA keys on default deployments. link:https://issues.redhat.com/browse/GITOPS-3042[GITOPS-3042]

Reviewer 2: ```suggestion
* Before this update, there was a mismatch in the RSA key for known hosts in the `argocd-ssh-known-hosts-cm` config map. This update fixes the issue by matching the RSA key with the upstream project. Now, you can use the default RSA keys on default deployments. link:https://issues.redhat.com/browse/GITOPS-3042[GITOPS-3042]
```

> * Before this update, reconciliation timeout setting in the `argocd-cm` configmap was not being correctly applied to the controller. This update fixes the issue by correctly reading and applying the reconciliation timeout setting from the `argocd-cm` configmap. Now, users can modify the reconciliation timeout from the `appSync` setting without a problem. link:https://issues.redhat.com/browse/GITOPS-2810[GITOPS-2810]

Reviewer 3: ```suggestion
* Before this update, reconciliation timeout setting in the `argocd-cm` config map was not being correctly applied to the Argo CD application controller. This update fixes the issue by correctly reading and applying the reconciliation timeout setting from the `argocd-cm` config map. Now, users can modify the reconciliation timeout from the `AppSync` setting without a problem. link:https://issues.redhat.com/browse/GITOPS-2810[GITOPS-2810]
```

Reviewer 3: The setting in the CR is shown as `AppSync`.

Reviewer 3: The names of the instances for all the above RNs were suggested based on the guideline:
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting
which was also pointed out by @Srivaralakshmi here:
https://github.com/openshift/openshift-docs/pull/62126#discussion_r1255517596

Author: Yes, I Missed that in my last update, and i was going to address in my next commit.

> * Before this update, reconciliation timeout setting in the `argocd-cm` config map was not being correctly applied to the Argo CD application controller. This update fixes the issue by correctly reading and applying the reconciliation timeout setting from the `argocd-cm` config map. Now, users can modify the reconciliation timeout from the `AppSync` setting without a problem. link:https://issues.redhat.com/browse/GITOPS-2810[GITOPS-2810]

Reviewer 2: ```suggestion
* Before this update, the reconciliation timeout setting in the `argocd-cm` config map was not being correctly applied to the Argo CD application controller resource. This update fixes the issue by correctly reading and applying the reconciliation timeout setting from the `argocd-cm` config map. Now, you can modify the reconciliation timeout value from the `AppSync` setting without a problem. link:https://issues.redhat.com/browse/GITOPS-2810[GITOPS-2810]
```
